File: лаба2/C++/laba2_3zadanie.cpp
#include <iostream>
using namespace std;

int count_dig(int num){//функция подсчета нечетных цифр в числе
    int count = 0;
    while (num > 0){
        int dig = num % 10;
        if(dig % 2 != 0){
            count++;
        }
        num /= 10;
    }
    return count;

}

int main(){
    int num, n;
    cin >> num;
    for (int i = 0; i < num; i++){
        cin >> n;
        cout << count_dig(n) << " ";
    }
    
}

File: лаба2/Go/laba2_3zadanie.go
package main

import "fmt"

// Считает количество нечётных цифр в числе
func count_dig(num int) int {
	count := 0
	for num > 0 {
		dig := num % 10
		if dig%2 != 0 {
			count++
		}
		num /= 10
	}
	return count
}

func main() {
	var num int
	fmt.Print("Введите количество чисел: ")
	fmt.Scan(&num) // Ввод количества чисел

	for i := 0; i < num; i++ {
		var n int
		fmt.Print("Введите число: ")
		fmt.Scan(&n) // Ввод числа
		fmt.Print(count_dig(n), "\n") // Вывод результата
	}
}

